src/java.base/share/classes/java/lang/invoke/MemberName.java

```diff
@@ -607,7 +575,7 @@ public MemberName(Method m, boolean wantSpecial) {
             }
             throw new LinkageError(m.toString());
         }
-        assert(isResolved() && this.clazz != null);
+        assert(isResolved());
```

Comment:
I don't see why this can be removed. Can you explain?

Reply:
the `if (clazz == null)` block above either returns or throws an error, so intellij was marking this as always true.